MotionPhotoReader/app/src/main/java/com/google/android/libraries/motionphotoreader/OutputSurface.java

```diff
@@ -0,0 +1,268 @@
+            // initialized
+            surfaceTexture = new SurfaceTexture(surfaceTextureHandle);
+            decodeSurface = new Surface(surfaceTexture);
+            surfaceTexture.setOnFrameAvailableListener(this);
+        });
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN_MR1)
+    private void eglSetup(int width, int height) {
+        renderHandler.post(() -> {
+            // Initialize EGL display
+            eglDisplay = EGL14.eglGetDisplay(EGL14.EGL_DEFAULT_DISPLAY);
+            if (eglDisplay == EGL14.EGL_NO_DISPLAY) {
+                throw new RuntimeException("Unable to get EGL14 display");
+            }
+
+            // Initialize EGL
+            int[] version = new int[2];
+            if (!EGL14.eglInitialize(eglDisplay, version, 0, version, 1)) {
+                eglDisplay = null;
+                throw new RuntimeException("Unable to initialize EGL14");
+            }
+
+            // Configure EGL
+            int[] numConfigs = new int[1];
+            EGLConfig[] configs = new EGLConfig[1];
+            int[] attributes = {
+                    EGL14.EGL_RENDERABLE_TYPE, EGL14.EGL_OPENGL_ES2_BIT,
+                    EGL14.EGL_RED_SIZE, 8,
+                    EGL14.EGL_GREEN_SIZE, 8,
+                    EGL14.EGL_BLUE_SIZE, 8,
+                    EGL14.EGL_ALPHA_SIZE, 8,
+                    EGL14.EGL_DEPTH_SIZE, 8,
+                    EGL14.EGL_STENCIL_SIZE, 0,
+                    EGL14.EGL_NONE
+            };
+            if (!EGL14.eglChooseConfig(
+                    eglDisplay,
+                    attributes, /* attrib_listOffset = */0,
+                    configs, /* configsOffset = */0,
+                    configs.length, numConfigs, /* num_configOffset = */ 0
+            )) {
+                throw new RuntimeException("eglChooseConfig failed");
+            } else if (numConfigs[0] == 0) {
+                throw new IllegalArgumentException("Could not find suitable EGLConfig");
+            }
+            eglConfig = configs[0];
+
+            // Initialize EGL context
+            eglContext = EGL14.eglCreateContext(
+                    eglDisplay,
+                    eglConfig,
+                    EGL14.EGL_NO_CONTEXT,
+                    new int[] {
+                            EGL14.EGL_CONTEXT_CLIENT_VERSION, 3,
+                            EGL14.EGL_NONE
+                    },
+                    /* offset = */ 0
+            );
+
+            // Make context current (with EGL_NO_SURFACE)
+            eglSurface = EGL14.EGL_NO_SURFACE;
+            if (!EGL14.eglMakeCurrent(eglDisplay, eglSurface, eglSurface, eglContext)) {
+                glError++;
+                Log.e(TAG, "Failed to make context current");
+            }
+
+            // Check for any errors
+            glError += GLES20.glGetError();
+        });
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.KITKAT)
+    public void setSurface(Surface surface) {
```

Comment:
you need to pass down the size of the Surface too, to set the correct viewport in OpenGL.